x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/date_index.test.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { setup, SetupResult, getProcessorValue } from './processor.helpers';
+
+const DATE_INDEX_TYPE = 'date_index_name';
+
+describe('Processor: Date Index', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup({
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+    testBed.component.update();
+    const {
+      actions: { addProcessor, addProcessorType },
+    } = testBed;
+    // Open the processor flyout
+    addProcessor();
+
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(DATE_INDEX_TYPE);
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" and "value" are required parameters
+    expect(form.getErrorsMessages()).toEqual([
+      'A field value is required.',
+      'A field value is required.',
+    ]);
+  });
+
+  test('saves with required field and date rounding parameter values', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Add "field" value (required)
+    form.setInputValue('fieldNameField.input', '@timestamp');
+
+    // Select second value for date rounding
+    form.setSelectValue('dateRoundingField', 's');
+
+    // Save the field
+    await saveNewProcessor();
+
+    const processors = await getProcessorValue(onUpdate, DATE_INDEX_TYPE);
+    expect(processors[0].date_index_name).toEqual({
+      field: '@timestamp',
+      date_rounding: 's',
+    });
+  });
+
+  test('allows optional parameters to be set', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+      find,
+      component,
+    } = testBed;
+
+    form.setInputValue('fieldNameField.input', 'field_1');
+
+    form.setSelectValue('dateRoundingField', 'd');
+
+    form.setInputValue('indexNamePrefixField.input', 'prefix');
+
+    form.setInputValue('indexNameFormatField.input', 'yyyy-MM');
+
+    await act(async () => {
+      find('dateFormatsField.input').simulate('change', [{ label: 'ISO8601' }]);
+    });
+    component.update();
+
+    form.setInputValue('timezoneField.input', 'EST');
```

Comment:
[nit] Looks like `EST` isn't a valid timezone supported by ES (related issue: https://github.com/elastic/kibana/issues/103368). So while this test passes and is working as expected, if the same data was used to make the request, the request itself would end up failing:

<img width="712" alt="Screen Shot 2021-06-29 at 2 20 49 PM" src="https://user-images.githubusercontent.com/5226211/123848088-4d3ebe00-d8e5-11eb-951c-7fc852d1677d.png">

Reply:
Fixed. I changed this to GMT which works with the API. Thanks for the catch.